debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgproto/PgProtoColumnValue.java

```diff
@@ -325,6 +325,9 @@ public Object asDefault(TypeRegistry typeRegistry, int columnType, String column
                 type.getOid() == typeRegistry.hstoreArrayOid()) {
             return asArray(columnName, type, fullType, connection);
         }
+        if (type.isArrayType() && type.getElementType().isEnumType()) {
+            return asArray(columnName, type, fullType, connection);
+        }
```

Comment:
I had to add this or else my test would just get an array of bytes. It looks like this file is responsible for the protobuf decoder? Do I need to do anything to make sure things work with native logical replication & wal2json?

Reply:
That's a good catch, it shouldn't go here for an array type in the first place. I've pushed a commit for rectifying this, fixing the `isArray()` method which didn't handle these types correctly.